SETUP.sh

```
@@ -81,27 +107,47 @@
+      log_warning "docker compose v2 not found. Falling back to 'docker-compose' if available."
+    fi
+    # Start Postgres service (idempotent)
+    if docker compose config >/dev/null 2>&1; then
+      docker compose up -d postgres
     else
-        log_warning "No database migrations found in src/db/migrations"
+      command -v docker-compose >/dev/null 2>&1 && docker-compose up -d postgres || true
     fi
+    # Wait for readiness
+    log_info "Waiting for PostgreSQL to become healthy..."
+    for i in {1..30}; do
```

Shellcheck (0.10.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)